gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/LaunchDagProc.java

```diff
@@ -63,8 +63,12 @@ protected Optional<Dag<JobExecutionPlan>> initialize(DagManagementStateStore dag
     try {
       FlowSpec flowSpec = dagManagementStateStore.getFlowSpec(FlowSpec.Utils.createFlowSpecUri(getDagId().getFlowId()));
       flowSpec.addProperty(ConfigurationKeys.FLOW_EXECUTION_ID_KEY, getDagId().getFlowExecutionId());
-      return this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec).toJavaUtil();
-    } catch (URISyntaxException | SpecNotFoundException | InterruptedException e) {
+      Optional<Dag<JobExecutionPlan>> dag = this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec).toJavaUtil();
+      if (dag.isPresent()) {
+        dagManagementStateStore.checkpointDag(dag.get());
+      }
+      return dag;
```

Comment:
What happens here if the dag is not present? E.g. if the dag fails to pass validation due to concurrency etc.

Reply:
it should just return `Optional.empty`, which `act` checks for just below

YMMV, but I find [this approach](https://github.com/apache/gobblin/pull/3942#discussion_r1589492017) to be a clearer signal that an alt. handling path was not missed

Reply:
I can simplify it to return optional.Empty explicitly. The approach you suggested doesn't work without explicitly surrounding the lambda with another try catch block (I can't add the exception to the existing try/catch block) which becomes slightly messy. wdyt?

Reply:
not sure if I'm missing something... why would you return `Optional.empty` explicitly?  this should be enough:
```
Optional<Dag<JobExecutionPlan>> dag = this.flowCompilationValidationHelper.createExecutionPlanIfValid(flowSpec).toJavaUtil();
```
it would add tautological boiler-plate to write something like:
```
if (dag.isPresent()) {
  ...
  return dag;
} else {
  return Optional.empty();
}
```
is that what you're suggesting?

seems as inadvisable as replacing:
```
return foo > bar;
```
with
```
return foo > bar ? true : false;
```

Reply:
Yea good pt, making that change isn't necessary.